src/utils/time.js

```diff
@@ -6,3 +10,54 @@ export function isValidDateString (value) {
     return false
   }
 }
+
+const mapQuasarTimeUnitToStelace = {
+  m: 'minutes',
+  h: 'hours',
+  d: 'days',
+  M: 'months'
+}
+
+function getQuasarTimeUnit (timeUnit) {
+  return mapQuasarTimeUnitToStelace[timeUnit]
+}
+
+export function convertEndDateFromAPIToUI (endDate, { startDate, timeUnit = 'd', includeEndDate = true } = {}) {
+  if (!includeEndDate) return endDate
+
+  const quasarTimeUnit = getQuasarTimeUnit(timeUnit)
+
+  if (!endDate) return endDate
+  const newEndDate = addToDate(new Date(endDate).toISOString(), { [quasarTimeUnit]: -1 }).toISOString()
+
+  // align the offset to start date to avoid non-integer duration
+  const differenceOffset = getOffsetDifference(startDate, newEndDate)
+  if (differenceOffset) {
+    return addToDate(newEndDate, { minutes: differenceOffset }).toISOString()
+  } else {
+    return newEndDate
+  }
+}
+
+export function convertEndDateFromUIToAPI (endDate, { startDate, timeUnit = 'd', includeEndDate = true } = {}) {
```

Comment:
`convertEndDateFromAPIToUI` and `convertEndDateFromUIToAPI` are very redundant.

Maybe you could create an object with two methods sharing most of their logic.